### ISSUE TITLE
Sparkle Binary Update Fails with Multiple Storyboards

### ISSUE THREAD

**Issue by Deadpikle** [sparkle-project/Sparkle#1103]
Awhile back, I refactored a rather-large program's Main.storyboard to use multiple storyboards via storyboard references. Works great, but ever since then, I've had to fight with the Sparkle BinaryUpdate software to generate working delta files. Sometimes it would work, but most of the time, the `BinaryDelta apply` step would fail with the error `Unable to extract file to [path]`.

I did some poking around today with the most recent source code. No progress in debugging, but I then attempted to do a straight extraction of the delta file to check its contents via `tar -xvf 0430to0431.delta` -- the first error that it omits is the same file that `BinaryDelta` fails on (which makes sense), and multiple other files fail to extract, too. All of the failed files are storyboard related (.nib). Perhaps this is indicative of a failure to generate an accurate patch file?

I'm willing to share delta files and perhaps even the software .app files (not source code!) with you for debugging, but would rather do so securely (at least via email) than on the open issue list. Please let me know how I can research this further on my end or work with you all to figure this one out. Thanks! :)

**Comment by pornel** [comment]
Have you tried the `generate_appcast` tool? Does it fail as well?

You can email me the files via github@pornel.net

**Comment by Deadpikle** [comment]
I haven't tried `generate_appcast` yet. Let me try that and get back to you in a few. Thanks.

**Comment by Deadpikle** [comment]
On first go-around, `generate_appcast` created a valid delta file (tested using a manual BinaryDelta just using the delta generated from `generate_appcast`). Strangely enough, I then went to go try to re-create a binary using `BinaryDelta`...and it worked! To make sure I wasn't insane, I then deleted all the app files and deltas and tried to make it work again. Subsequent attempts to create a working delta with `BinaryDelta` have failed. Very odd.

Expect an email in your inbox with links to files in a few minutes.
